docs/user/bots.rst

```diff
@@ -2611,6 +2611,75 @@ Use the following command to validate your sieve files:
    optional arguments:
      -h, --help  show this help message and exit
 
+Splunk saved search
+^^^^^^^^^^^^^^^^^^^
+
+**Information**
+
+* `name`: splunk_saved_search
+* `lookup`: splunk database
+* `public`: no
+* `cache (redis db)`: none
+* `description`: Enrich an event from Splunk search results.
+
+**Configuration Parameters**
+
+* **HTTP parameters** (see above)
+* `auth_token`: String, Splunk API authentication token
+* `url`: String, base URL of the Splunk REST API
+* `retry_interval`: Integer, optional, default 5, number of seconds to wait between attempts to retrieve search results
+* `saved_search`: String, name of Splunk saved search to run
+* `search_parameters`: Array of string->string, optional, default ``{}``, IntelMQ event fields containing the data to search for, mapped to parameters of the Splunk saved search. Example:
+
+  .. code-block:: json
+
+     "search_parameters": {
+         "source.ip": "ip"
+     }
+
+* `result_fields`: Array of string->string, optional, default ``{}``, Splunk search result fields mapped to IntelMQ event fields to store the results in. Example:
+
+  .. code-block:: json
+
+     "result_fields": {
+         "username": "source.account"
+     }
+
+* `not_found`: List of strings, default ``[ "warn", "send" ]``, what to do if the search returns zero results. Valid values are:
+
+  * `warn`: log a warning message
+  * `send`: send the event on unmodified
+  * `drop`: drop the message
+
+    * `send` and `drop` are mutually exclusive
+
+* `duplicates`: List of strings, default ``[ "warn", "use_first", "send" ]``, what to do if the search returns more than one result. Valid values are:
```

Comment:
I cannot thing of a better single word, and would prefer not to have to extend configuration parameters to phrases. But I'm not a native speaker, so if you have a better suggestion, go ahead.

There is [a "head" command](https://docs.splunk.com/Documentation/Splunk/latest/SearchReference/Head) that limits the return to a number of results, but I had never heard of it before your comment made me go look. My impression is that idiomatic Splunk would limit the number of results by constraining the time window of the search (which cannot be done through the API - [the documentation for running saved searches](https://docs.splunk.com/Documentation/Splunk/latest/SearchReference/Savedsearch) references the "time picker", a web UI widget with no API counterpart). My impression is also that Splunk is heavily weighted towards interactive use in the web UI, with the REST API being at best an afterthought.

I would prefer not having to limit my search results when writing saved searches, since that makes them less useful for interactive use or creating reports. It would probably be possible to have a configuration parameter to let the bot add a `| head 1` command at the end of the query.

Reply:
I tested, and blindly adding `| head 1` at the end of the query works, so I added an option to specify that to guarantee there will be no duplicate results. That way, you can have a saved search that returns an arbitrary number of results interactively, but just one when run from the bot.

Reply:
> I cannot thing of a better single word, and would prefer not to have to extend configuration parameters to phrases. But I'm not a native speaker, so if you have a better suggestion, go ahead.

What do you think about `duplicate_handling`?



> I tested, and blindly adding `| head 1` at the end of the query works, so I added an option to specify that to guarantee there will be no duplicate results. That way, you can have a saved search that returns an arbitrary number of results interactively, but just one when run from the bot.

Nice!

Reply:
I find `multiple_result_handling` to be clearer for a human reading the configuration, but less elegant, especially when used as a variable name in the code. But I added a commit changing the name, so you can see which one you prefer.